### PR TITLE
Executable jar

### PR DESCRIPTION
**Pull request by jdigger**
Implements GH-2

Start of making this more useful than the shell script. Per the updated README, it's fully self-contained now, running just like the shell script as a simple executable. The only pre-req beyond the shell script is the JRE.

Please review, @masroormohammed, @sagarsane, @jbornemann 

cc: @jazeren1 @cculb

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/TWCable/grabbit-cli/pull/8" target="_blank">ReviewNinja

**Comment by jbornemann** [comment]
Test/CR +1

I'm am curious what the evolution of this project will be; Specifically, how it will evolve to become more useful than our Grabbit bundled shell launcher script.

**Comment by jdigger** [comment]
Can you give me a ninja star? ⭐  :-) (Click on the ninja girl above.) Testing out using ReviewNinja for improving the CR process...

**Comment by jdigger** [comment]
Some of the advantages compared to the shell script:
1. Runs on Windows just as easily as Linux/OSX
2. Much easier to write readable/maintainable code
3. Much easier to do proper error handling
4. Much easier to write tests for
   - Besides the obvious value of having tests for regression/validation, there's also a significant documentation value to them
5. Can start acting as the basis for other forms of UI
6. More easily extensible to other workflows for working with Grabbit without having to copy & paste
   - It's easier (and more idiomatic) to extract out common functionality

Disadvantages compared to the shell script:
1. Requires the JRE

File: build.gradle [hunk 1]
```diff
@@ -4,46 +4,40 @@ buildscript {
...
-
+//noinspection GroovyAssignabilityCheck
 dependencies {
-    compile "org.codehaus.groovy:groovy-all:2.4.3"
+    compile "org.codehaus.groovy:groovy-all:2.4.6"
     compile 'commons-cli:commons-cli:1.2'
 
     // @Nonnull @Nullable etc annotations
-    compile "com.google.code.findbugs:jsr305:2.0.0"
+    // compile "com.google.code.findbugs:jsr305:2.0.0"
```
Comment (sagarsane) [inline build.gradle]: I suppose we can remove this line.

Comment (sagarsane) [inline build.gradle]: !fix

Comment (jdigger) [inline build.gradle]: !fixed It got removed later as part of the `to-java` branch (in favor of the Checker Framework), but I've removed in this branch as well.

Comment (sagarsane) [inline build.gradle]: AH ok

Comment (jbornemann) [inline build.gradle]: !fix

Comment (jbornemann) [inline build.gradle]: !fixed

Comment (jbornemann) [inline build.gradle]: !resolve

Comment (jbornemann) [inline build.gradle]: !fixed

**Comment by jbornemann** [comment]
A bit confused. The "Add ninja star" button on this tool seems to not do anything. I click, and get no feedback

**Comment by jbornemann** [comment]
1, 5, and 6 I see as exciting possibilities; however maybe more as a substitute for advanced use, rather than an all use-case substitute. When I as a lone-user, use Grabbit, I want to get going as fast as possible; I don't need to worry about multiple environments, or managing environment files just to get started. I type, ./grabbit.sh, work through a couple of prompts, and I'm off. Maybe this project could be that as well one day, but I think think grabbit.sh still has it's place for lightweight, kick-start use. Of course, more robust use such as CI paints a much different story in terms of interactivity, as well as other features. We could see a lot from this project here in that regard.

For test coverage, and maintainable code; this will become more and more important as we see higher development traffic in these UI areas. Grabbit command-line has maybe been touched a handful of times over the years, and for mostly trivial reasons - partly because what it does is trivial in itself. This sort of leads me to my curiosity of where the larger CLI project is going in terms of features - environment files are nice, but you seem to hint at the possibility of maybe something more - more advanced graphical reporting, etc - not to completely speak for you, or others.

**Pull request merged**